activemq-broker/src/main/java/org/apache/activemq/broker/region/Queue.java

```diff
@@ -2210,6 +2210,16 @@ private PendingList doActualDispatch(PendingList list) throws Exception {
                         // no further dispatch of list to a full consumer to
                         // avoid out of order message receipt
                         fullConsumers.add(s);
+
+                        //For full consumers we need to mark that they are slow and
+                        // then call the broker.slowConsumer() hook if implemented
+                        if (s instanceof PrefetchSubscription) {
```

Comment:
Should we honor the isAdvisoryForSlowConsumer config flag? Here or maybe better in the AdvisoryBroker.

Reply:
You can't really add that check here because that the callback could be used for many different things. Advisories is just one use case and implementation.